runtime/amd64.S

```diff
@@ -1341,6 +1363,9 @@ CFI_STARTPROC
         movq    Caml_state(exn_handler), %r10
         movq    %rsp, Stack_sp(%rcx)
         movq    %r10, Stack_exception(%rcx)
+    /* Save local allocations state that the OCaml code might have modified */
+        movq    Caml_state(local_sp), %r10
+        movq    %r10, Stack_local_sp(%rcx)
```

Comment:
Why is this necessary here?

Reply:
We're about to switch `Caml_state->current_stack` to the new fiber, so we need to preemptively update the current stack's `local_sp` (same as `SWITCH_OCAML_STACKS`)